src/cljs/main/broadfcui/page/workspace/method_configs/tab.cljs

```diff
@@ -33,44 +27,29 @@
        (cond
          error-message (style/create-server-error-message error-message)
          configs
-         [table/Table
-          {:empty-message "There are no method configurations to display."
-           :toolbar
-           (add-right
+         (ws-common/method-config-selector
+          {:configs configs
+           :render-name (fn [config]
+                          (style/create-link {:text (:name config)
+                                              :href (nav/get-link :workspace-method-config
+                                                                  (:workspace-id props)
+                                                                  (ws-common/config->id config))}))
+           :toolbar-items
+           [flex/spring
             [comps/Button
              {:text "Import Configuration..."
               :disabled? (case locked?
                            nil "Looking up workspace status..."
                            true "This workspace is locked."
                            false)
               :onClick #(modal/push-modal
-                         [comps/OKCancelForm
-                          {:header "Import Method Configuration"
-                           :content
-                           [:div {:style {:backgroundColor "white" :padding "1rem"}}
-                            [MethodConfigImporter
-                             {:workspace-id (:workspace-id props)
-                              :after-import (fn [{:keys [config-id]}]
-                                              (modal/pop-modal)
-                                              ((:on-config-imported props) config-id))}]]}])}])
-           :columns
-           [{:header "Name" :starting-width 240 :as-text :name :sort-by :text
-             :content-renderer
-             (fn [config-id]
-               (style/create-link {:text (:name config-id)
-                                   :href (nav/get-link :workspace-method-config
-                                                       (:workspace-id props)
-                                                       config-id)}))}
-            {:header "Root Entity Type" :starting-width 140}
-            {:header "Method" :starting-width 800
-             :content-renderer (fn [fields] (apply style/render-entity fields))}]
-           :data configs
-           :->row (fn [config]
-                    [(config->id config)
-                     (config "rootEntityType")
-                     (mapv #(get-in config ["methodRepoMethod" %])
-                           ["methodNamespace" "methodName" "methodVersion"])])}]
-         :else [:div {:style {:textAlign "center"}} [comps/Spinner {:text "Loading configurations..."}]])))
+                         [ConfigImporter
+                          {:workspace-id (:workspace-id props)
+                           :after-import (fn [{:keys [config-id]}]
```

Comment:
1) I think this is the right place to handle this (I especially like modal push and pop to be handled by the same component)...